common/src/chain/transaction/signature.rs

```diff
@@ -0,0 +1,254 @@
+    witness.verify_signature(outpoint_destination, &sighash)?;
+    Ok(())
+}
+
+pub fn verify_signature(
+    outpoint_destination: &Destination,
+    tx: &Transaction,
+    input_num: usize,
+) -> Result<(), TransactionSigError> {
+    let target_input = tx
+        .get_inputs()
+        .get(input_num)
+        .ok_or_else(|| TransactionSigError::InvalidInputIndex(input_num, tx.get_inputs().len()))?;
+    let input_witness = target_input.get_witness();
+    match input_witness {
+        inputsig::InputWitness::NoSignature(_) => {
+            return Err(TransactionSigError::SignatureNotFound)
+        }
+        inputsig::InputWitness::Standard(witness) => {
+            verify_standard_input_signature(outpoint_destination, witness, tx, input_num)?
+        }
+    }
+    Ok(())
+}
+// TODO: write tests
+
+#[cfg(test)]
+mod test {
+    use std::vec;
+
+    use crypto::key::{KeyKind, PrivateKey};
+
+    use super::{
+        inputsig::{InputWitness, StandardInputSignature},
+        sighashtype::SigHashType,
+    };
+    use crate::{
+        chain::{
+            signature::verify_signature, Destination, Transaction, TransactionCreationError,
+            TxInput, TxOutput,
+        },
+        primitives::{Amount, Id, H256},
+    };
+
+    fn generate_tx(
+        private_key: PrivateKey,
+        sighash_type: SigHashType,
+        dest: Destination,
+    ) -> Result<Transaction, TransactionCreationError> {
+        let mut tx = Transaction::new(
+            0,
+            vec![
+                TxInput::new(
+                    Id::<Transaction>::new(&H256::zero()).into(),
+                    0,
+                    InputWitness::NoSignature(None),
+                ),
+                TxInput::new(
+                    Id::<Transaction>::new(&H256::random()).into(),
+                    1,
+                    InputWitness::NoSignature(None),
+                ),
+            ],
+            vec![TxOutput::new(Amount::from_atoms(100), dest.clone())],
+            0,
+        )?;
+
+        let signature = StandardInputSignature::produce_signature_for_input(
+            &private_key,
+            sighash_type,
+            dest.clone(),
+            &tx,
+            0,
+        )
+        .unwrap();
+
+        tx.update_witness(0, InputWitness::Standard(signature.clone())).unwrap();
+        Ok(tx)
+    }
+
+    fn sign_tx(
+        tx: &mut Transaction,
+        private_key: &PrivateKey,
+        sighash_type: SigHashType,
+        dest: Destination,
+    ) {
+        for i in 0..tx.get_inputs().len() {
+            let input_sign = StandardInputSignature::produce_signature_for_input(
+                private_key,
+                sighash_type,
+                dest.clone(),
+                &tx,
+                0,
+            )
+            .unwrap();
+            tx.update_witness(i, InputWitness::Standard(input_sign)).unwrap();
+        }
+    }
+
+    #[test]
+    #[allow(clippy::eq_op)]
+    fn sign_and_verify() {
```

Comment:
Test names should be meaningful. Please call this something like `sign_and_verify_different_sighash_types()`.

Reply:
Gotcha! Fixed!